hbase-mapreduce/src/test/java/org/apache/hadoop/hbase/mapreduce/TestHFileOutputFormat2.java

```diff
@@ -657,9 +657,9 @@ private void doIncrementalLoadTest(boolean shouldChangeRegions, boolean shouldKe
       assertEquals("HFOF should not touch actual table", 0, util.countRows(tableSingle));
```

Comment:
this test method `doIncrementalLoadTest` is already parameterized. I wonder if we could add a test for the new `writeToTableWithNamespace` mode. Basically add a:

```
if (writeMultipleTables && writeToTableWithNamespace) {
  testDir = new Path(testDir, "default");
}
```

Sort of like you had before, but now wrapped in a new boolean. We can then add a new `@Test` method which passes true for that so we can verify that we can write to the correct output dirs for both the old and new structure, when configured appropriately.

Reply:
Same in TestCellBasedHFileOutputFormat2

Reply:
Yea, I will cover the new flows.